llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9151,9 +9011,12 @@ void LoopVectorizationPlanner::adjustRecipesForReductions(
               ? new VPInstruction(Instruction::Select, {Cond, Red, PhiR}, FMFs)
               : new VPInstruction(Instruction::Select, {Cond, Red, PhiR});
       Result->insertBefore(&*Builder.getInsertPoint());
-      Red->replaceUsesWithIf(
-          Result->getVPSingleValue(),
-          [](VPUser &U, unsigned) { return isa<VPLiveOut>(&U); });
+      Red->replaceUsesWithIf(Result->getVPSingleValue(),
```

Comment:
Would it be better to introduce these `Select`'s for non in-loop reductions under tail folding, along with the introduction of `ComputeReductionResult`? Rather than first introducing the latter and then searching to replace its operand by the former.

Note that these Selects originally had to be introduced inside the loop, but can now be placed in the middle block alongside ComputeReductionResult.

nit (unrelated to this patch): is the enclosing `for` statement mis-indented, along with its preceding `Builder.setInsertPoint(&*LatchVPBB->begin());` statement?